docs/tutorial/creating_cli.md

```diff
@@ -0,0 +1,635 @@
+
+It makes little sense to define the behaviour of the CLI in terms of an interface, as there are only a few ways of interacting with the user via a console. So we're going to contain all the logic in a class that has the `ContactManager` as a dependency.  
+
+`~/src/main/shellfish/contacts/cli/Cli.scala`
+```scala 3
+class Cli private (cm: ContactManager): 
+  def addCommand: IO[Unit] = ...
+```
+
+The first function is going to be the `addCommand`. The idea is to ask the user step by step for the information of the contact and then add it to the contact list:
+
+`~/src/main/shellfish/contacts/cli/Cli.scala`
+```scala 3
+def addCommand: IO[Unit] =
+  for 
+    username    <- IO.println("Enter the username: ") >> IO.readLine
+    firstName   <- IO.println("Enter the first name: ") >> IO.readLine
+    lastName    <- IO.println("Enter the last name: ") >> IO.readLine
+    phoneNumber <- IO.println("Enter the phone number: ") >> IO.readLine
+    email       <- IO.println("Enter the email: ") >> IO.readLine
+
+    contact = Contact(username, firstName, lastName, phoneNumber, email)
+    
+    _ <- cm.addContact(contact)
+      .flatMap(username => IO.println(s"Contact $username added"))
+      .handleErrorWith:
+        case ContactFound(username) =>
+          IO.println(s"Contact $username already exists")
+        case e =>
+          IO.println(s"An error occurred: \n${e.printStackTrace()}")
+    
+  yield ()
+```
+
+The first five `IO.readLine` calls in `for` asks the fields to save the contact. Then we create the `Contact` object with the information provided. Next, we call the `addContact` method of the `ContactManager` and handle the errors. If the contact already exists, we print a message to the user. If another error occurs, we print the stack trace of the error to see what went wrong.  
+
+Now the `removeCommand` function:
+
+`~/src/main/shellfish/contacts/cli/Cli.scala`
+```scala 3 
+def removeCommand(username: Username): IO[Unit] =
+  cm.removeContact(username) >> IO.println(s"Contact $username removed")
+```
+It simply calls the `removeContact` function and prints a message to the user. The function doesn't check if the contact exists and will delete any matching username anyway, but you can add this logic if you want.
+
+The `searchIdCommand` function is also straight forward, matches the `Options` returned by the `searchId` method and prints the user information if it exists:
+
+`~/src/main/shellfish/contacts/cli/Cli.scala`
+```scala 3
+def searchIdCommand(username: Username): IO[Unit] =
+  for
+    contact <- cm.searchId(username)
+    _ <- contact match 
+      case Some(c) => IO.println(c.show)
+      case None    => IO.println(s"Contact $username not found")
+  yield ()
+```
+
+Like the previous `searchX` variants, they’re fairly similar, so we will only show one of them.
+In this case, the `searchEmailCommand` one:
+
+`~/src/main/shellfish/contacts/cli/Cli.scala`
+```scala 3
+def searchEmailCommand(email: Email): IO[Unit] =
```

Comment:
I would recommend this style for the path specification
```suggestion
```scala 3
// src/main/shellfish/contacts/cli/Cli.scala
def searchEmailCommand(email: Email): IO[Unit] =
```